test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -942,14 +937,20 @@ void UcxConnection::request_completed(ucx_request *r)
 void UcxConnection::handle_connection_error(ucs_status_t status)
 {
     if (UCS_STATUS_IS_ERR(_ucx_status)) {
+        assert(_establish_cb == NULL);
         return;
     }
 
     UCX_CONN_LOG << "detected error: " << ucs_status_string(status);
     _ucx_status = status;
 
     /* the upper layer should close the connection */
-    _context.handle_connection_error(this);
+    if (is_established()) {
```

Comment:
isn't this some duplication of https://github.com/openucx/ucx/pull/6338/commits/d71ac19d5da84c5d355e3a5e0b46528c1f837d7a#diff-d3eefc2ff6f38844ae09889a04bb916b5433731d5780f8a10bd9d6bf7862ab41R370?

Reply:
no but I agree - naming looks confusing...